### DISCUSSION TITLE
blog/validation-with-mediatr-pipeline-behavior-and-fluentvalidation/

### DISCUSSION THREAD

**Discussion by giscus** [codewithmukesh/discussions#13] [General]
# blog/validation-with-mediatr-pipeline-behavior-and-fluentvalidation/

Learn how to use MediatR Pipeline Behavior to implement Validations with FluentValidation in ASP.NET Core applications. This pattern is extremely powerful when clubbed with IExceptionHandler, and can help you build super clean solutions with an elegant way to handle Validations!

https://codewithmukesh.com/blog/validation-with-mediatr-pipeline-behavior-and-fluentvalidation/?utm_source=linkedin

**Comment by mallidasari** [comment]
very clear and precise article. Keep sharing.

**Comment by sofia199999** [comment]
It is very well information about of this page. All people always want to know all about things that you have been described.
<a href="https://bermudaunicorn.com/collection/city-view/”>City View</a>

**Comment by sofia199999** [comment]
It is very well information about of this page. All people always want to know all about things that you have been described.
<ahref="https://bermudaunicorn.com/collection/city-view/”>City-View

**Comment by Hamnava** [comment]
It was really useful, and I enjoyed reading your article.
Thanks for sharing!

**Comment by naufilshaikh09** [comment]
Another great article.

**Comment by terius** [comment]
If I create two different validators for the same operation class, the resulting error messages will be duplicated. How can I resolve this?

**Comment by terius** [comment]
public class CreateProductCommandValidator : AbstractValidator<CreateProductCommand>
    {
        public CreateProductCommandValidator()
        {
            RuleFor(p => p.Name).NotEmpty().MinimumLength(4);
            RuleFor(p => p.Price).GreaterThan(0);
        }
    }

    public class CreateProductCommandValidator2 : AbstractValidator<CreateProductCommand>
    {
        public CreateProductCommandValidator2()
        {
            RuleFor(p => p.Description).NotEmpty().MaximumLength(10);
        }
    }

result：
 {
        "type": "https://tools.ietf.org/html/rfc7231#section-6.5.1",
        "title": "one or more validation errors occurred.",
        "status": 400,
        "instance": "/products",
        "errors": [
            "'Name' 必须大于或等于4个字符。您输入了2个字符。",
            "'Price' 必须大于 '0'。",
            "'Description' 必须小于或等于10个字符。您输入了26个字符。",
            "'Name' 必须大于或等于4个字符。您输入了2个字符。",
            "'Price' 必须大于 '0'。",
            "'Description' 必须小于或等于10个字符。您输入了26个字符。"
        ]
    },

**Reply by CarlosPavajeau** [reply]
On  `GlobalExceptionHandler` convert error list to frozen set

```c_sharp
var validationErrors = fluentException.Errors
                .Select(error => error.ErrorMessage)
                .ToFrozenSet();

problemDetails.Extensions.Add("errors", validationErrors);
```
